RELEASE.md: Mention period of time for rc (#823)
kubernetes/kube-state-metrics: RELEASE.md

Maintaining the release branches for older minor releases happens on a best effort basis.

## Prepare your release

* Bump the version in the `VERSION` file in the root of the repository.
* Make a PR to update:

>   * kube-state-metrics image tag for both `quay.io` and `staging-k8s.gcr.io`

Reviewer: Let's add the full stop as well.
```suggestion
  * kube-state-metrics image tag for both `quay.io` and `staging-k8s.gcr.io`.
```

    [BUGFIX]
    ```
    * All lines should be full sentences

>   * kube-state-metrics image tag used in Kubernetes deployment yaml config

Reviewer: ```suggestion
  * kube-state-metrics image tag used in Kubernetes deployment yaml config.
```

* Cut the new release branch, i.e., `release-1.2`, or merge/cherry-pick changes onto the minor release branch you intend to tag the release on
* Cut the new release tag, i.e., `v1.2.0-rc.0`
* Ping Googlers(@loburm/@piosz) to build and push newest image to `k8s.gcr.io` (or to `staging-k8s.gcr.io` in case of release candidates)
* Build and push newest image to `quay.io`(@brancz)

## Stable release

> First a release candidate should be cut, e.g. `v.1.2.3-rc.0` should be cut and if after a period of 7 days no bugs or issues were reported with the release candidate, a stable release can be cut.

Reviewer: To make this more illustrative, let's add the `v1.2.3` as an example for a stable release.

> First a release candidate should be cut, e.g. `v.1.2.3-rc.0` should be cut and if after a period of 7 days no bugs or issues were reported with the release candidate, a stable release can be cut.

Reviewer: ```suggestion
First a release candidate (e.g. `v1.2.3-rc.0`) should be cut and if after a period of 7 days no bugs or issues were reported with the release candidate, a stable release can be cut.
```

Reviewer: Let's break this down instead of having a rather heavy sentence. 

The second sentence could be "If no bugs or issues were reported for a period of 7 days after publishing the release candidate, a stable release can be cut."
